embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/LogOrchestrator.kt

```diff
@@ -0,0 +1,78 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import androidx.annotation.VisibleForTesting
+import io.embrace.android.embracesdk.comms.delivery.DeliveryService
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.worker.ScheduledWorker
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class LogOrchestrator(
+    private val logOrchestratorScheduledWorker: ScheduledWorker,
+    private val clock: Clock,
+    private val sink: LogSinkImpl,
```

Comment:
This should be the interface `LogSink` instead so we can provide a fake impl for tests?